components/ThemeSwitch.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import { useTheme } from "next-themes";
+
+export const ThemeSwitch = () => {
+  const [mounted, setMounted] = useState(false);
+
+  const { theme, setTheme } = useTheme();
+
+  useEffect(() => {
+    setMounted(true);
```

Comment:
The mounting logic can be moved to a hook, but it's not necessary and more of a cosmetic change.